docs/source/notebooks/wps_decompose_flow_vectors_rename.ipynb

```diff
@@ -0,0 +1,233 @@
+    "    ]\n",
+    "    output_northward = [\n",
+    "        y_magnitude \n",
+    "        for subarray in output_data.variables[\"northward_Flow_Direction\"] \n",
+    "        for y_magnitude in subarray \n",
+    "        if y_magnitude != \"masked\"\n",
+    "    ]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check if input and output data sizes are matching\n",
+    "\n",
+    "assert len(input_data) == len(output_eastward)\n",
+    "assert len(output_eastward) == len(output_northward)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check if input and output outlet positions are matching\n",
+    "\n",
+    "outlets = [i for i in range(len(input_data)) if input_data[i] == 9]\n",
+    "\n",
+    "vector_addition = [output_eastward[i]**2 + output_northward[i]**2 for i in range(len(input_data)) if i in outlets]\n",
+    "expected_magnitudes = [0.0 for i in range(len(outlets))]\n",
+    "\n",
+    "assert vector_addition == expected_magnitudes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check if every output non-outlet vector magnitude eqauls to 1\n",
+    "\n",
+    "non_outlets = [i for i in range(len(input_data)) if input_data[i] != 9]\n",
+    "\n",
+    "vector_addition = [round(output_eastward[i]**2 + output_northward[i]**2, 4) for i in range(len(input_data)) if i in non_outlets]\n",
+    "expected_magnitudes = [1.0 for i in range(len(non_outlets))]\n",
+    "\n",
+    "assert vector_addition == expected_magnitudes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check output x and y vectors\n",
+    "\n",
+    "output_x_vectors = [round(output_eastward[i], 4) for i in non_outlets]\n",
+    "expected_x_vectors = [round(math.cos((input_data[i]-3) * math.pi/4 * -1), 4) for i in non_outlets]\n",
+    "assert output_x_vectors == expected_x_vectors\n",
+    "\n",
+    "output_y_vectors = [round(output_northward[i], 4) for i in non_outlets]\n",
+    "expected_y_vectors = [round(math.sin((input_data[i]-3) * math.pi/4 * -1), 4) for i in non_outlets]\n",
+    "assert output_y_vectors == expected_y_vectors"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.6.9"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 4
+}
```

Comment:
While I appreciate the diligent work that you have put into these assertion checks, I find them very dense. It is difficult to understand what exactly they are doing and without the comments it would be even more challenging.

We know that `ce-dataprep` is extensively tested, which means we can assume the results we get from running the modules are accurate. All we have to do is make sure `thunderbird` produced what we were expecting. My recommendation would be to keep the first two assertion cells and simplify the second one if possible.